src/libertem/io/dataset/hdf5.py

```diff
@@ -160,17 +160,26 @@ def _do_detect():
         # try to guess the hdf5 dataset path:
         try:
             datasets = executor.run_function(_get_datasets, path)
-            largest_ds = sorted(datasets, key=lambda i: i[1], reverse=True)[0]
-            name, size, shape, dtype = largest_ds
+            datasets_list = sorted(datasets, key=lambda i: i[1], reverse=True)
+            ds_paths = [ds_path[0] for ds_path in datasets_list]
+            # options for the semantic-ui dropdown
+            dataset_paths = [{"key": dpath, "text": dpath, "value": dpath} for dpath in ds_paths]
```

Comment:
I would prefer if the "formatting" into key/text/value would happen in the TypeScript code - for usage from Python this is redundant information, it's just there because of how the dropdown works

Reply:
Fixed!